src/QirRuntime/lib/Tracer/tracer.cpp

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <assert.h>
+#include <sstream>
+
+#include "tracer.hpp"
+
+using namespace std;
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    thread_local std::shared_ptr<CTracer> tracer = nullptr;
+    std::shared_ptr<CTracer> CreateTracer()
+    {
+        tracer = std::make_shared<CTracer>();
+        return tracer;
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // CTracer's ISumulator implementation
+    //------------------------------------------------------------------------------------------------------------------
+    IQuantumGateSet* CTracer::AsQuantumGateSet()
+    {
+        return nullptr;
+    }
+    IDiagnostics* CTracer::AsDiagnostics()
+    {
+        return nullptr;
+    }
+    Qubit CTracer::AllocateQubit()
+    {
+        size_t qubit = qubits.size();
+        qubits.push_back({});
+        return reinterpret_cast<Qubit>(qubit);
+    }
+    void CTracer::ReleaseQubit(Qubit /*qubit*/)
+    {
+        // nothing for now
+    }
+    std::string CTracer::QubitToString(Qubit q)
+    {
+        size_t qubitIndex = reinterpret_cast<size_t>(q);
+        const QubitState& qstate = this->UseQubit(q);
+
+        stringstream str(qubitIndex);

```

Comment:
This is causing the build to fail on linux.
```suggestion
        stringstream str(std::to_string(qubitIndex));
```

Reply:
Thank you! (Why, oh why, cannot Windows and Linux do the same thing here?)